Add Dockerfile and VSCode devcontainer

Islandman93 opened the pull request:
Depends on #10 to be merged first.

Islandman93 committed:
Dockerfile & vscode development container

Islandman93 committed:
Install ViZDoom locally in the dockerfile

Islandman93 commented on deps/build.jl:
```diff
@@ -1,21 +1,21 @@
 cd(() -> run(cmake_cmd), vizdoom_dir)
-cd(() -> run(`make`), vizdoom_dir)
\ No newline at end of file
+cd(() -> run(`make`), vizdoom_dir)
```
Ideally update to make -j `nproc`

findmyway commented:
I've sent you an invitation to maintain this project. So feel free to merge this PR and tag a new version when you think it is ready.

Islandman93 committed:
Add nproc to make for faster builds

Islandman93 committed:
Add docker user for security & mount permission issues
Mount X11 for devcontainer visualization
Devcontainer to use the same code point as the Dockerfile

Islandman93 committed:
Add some details to the README for docker

Islandman93 commented:
This PR doesn't modify any user facing code so I'm keeping the version number at v0.1.1
